plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -1126,4 +1124,29 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
     {
         return ImmutableSet.of();
     }
+
+    static class StatsRecordingAsyncHandler<Request extends AmazonWebServiceRequest, Result>
+            implements AsyncHandler<Request, Result>
+    {
+        private final GlueMetastoreStats stats;
+        private final long startTimeInMillis;
+
+        public StatsRecordingAsyncHandler(GlueMetastoreStats stats, long startTimeInMillis)
+        {
+            this.stats = stats;
+            this.startTimeInMillis = startTimeInMillis;
+        }
+
+        @Override
+        public void onError(Exception e)
+        {
+            stats.getBatchUpdatePartition().recordCall(System.currentTimeMillis() - startTimeInMillis, true);
```

Comment:
You should use different `GlueMetastoreApiStats` instance for different places when you use `StatsRecordingAsyncHandler`. Not `stats.getBatchUpdatePartition()` all the time.
Change parametrization of `StatsRecordingAsyncHandler` to use `GlueMetastoreApiStats` instead of `GlueMetastoreStats`

Reply:
yes definitely, I forgot to change that